packages/python/accern_data/accern_data.py

```diff
@@ -425,39 +492,54 @@ def _process_date(
             output_path: str,
             output_pattern: Optional[str],
             *,
-            is_first_day: bool,
-            progress_bar: ProgressBar) -> None:
+            is_first_time: bool,
+            progress_bar: ProgressBar) -> bool:
         self._params["date"] = cur_date
         first = True
         for res in self._scroll("1900-01-01"):
             is_empty = False
             if first:
                 self.get_mode().init_day(
-                    cur_date, output_path, output_pattern, is_first_day)
+                    cur_date, output_path, output_pattern, is_first_time)
                 first = False
+
             if isinstance(res, pd.DataFrame) and res.empty:
                 is_empty = True
             elif isinstance(res, list) and len(res) == 0:
                 is_empty = True
+
+            if is_first_time and not is_empty:
+                is_first_time = False
+
             if not is_empty:
                 self.get_mode().add_result(res)
                 progress_bar.update(len(res))
 
         if not first:
             self.get_mode().finish_day()
+        return is_first_time
 
     def download_range(
             self,
             start_date: str,
             output_path: Optional[str] = None,
             output_pattern: Optional[str] = None,
             end_date: Optional[str] = None,
+            mode: Optional[str] = None,
+            split_dates: bool = False,
+            filters: Optional[FiltersType] = None,
             verbose: bool = False) -> None:
         global VERBOSE
         VERBOSE = verbose
         if output_path is None:
             output_path = "./"
         os.makedirs(output_path, exist_ok=True)
+        if mode is None:
+            self.get_mode()
+        else:
+            self.set_mode(mode=cast(MODE, mode), split_dates=split_dates)
+        if filters is not None:
+            self.get_filters().update(filters)
```

Comment:
don't update the filters -- use them locally

Reply:
filters are used by internal functions i.e. for getting no of signals and downloading them.

passing them through functions can be done, but may look messy

Reply:
pass those things as arguments or create a copy of the object on change